### PR TITLE
Add Gmail Subject DEVEX-234

### PR DESCRIPTION
**Pull request by E-B-Smith**
Added some methods for setting the 'Subject' field for activities.

This didn't fix Gmail, but it won't hurt. It may help other activity extensions though.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/c4a08d815bcaafc35de319aa0e713540/c4a08d815bcaafc35de319aa0e71367f)
---
*Duration*: 17 minutes and 57 seconds
*Result*: **Build failed 2 tests** out of 227

File: Branch-SDK/Branch-SDK/BranchShareLink.m [hunk 1]
```diff
@@ -176,11 +189,10 @@ - (void) presentActivityViewControllerFromViewController:(UIViewController*_Null
         
     }
 
-    if (self.linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT]) {
+    NSString *emailSubject = self.linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT];
```
Comment (sojanpr) [inline Branch-SDK/Branch-SDK/BranchShareLink.m]: Adding it in control param will add it to the deep link data. Do we have a way someone can  specify the subject with out adding it to the link data?

**Comment by E-B-Smith** [comment]
* Added an option to set an email subject without adding it as a link parameter.
      - This is similar to the existing Android option.
@sojanpr

**Comment by BranchMacMini** [comment]
Result of [Integration 3](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/c4a08d815bcaafc35de319aa0e713540/b1af2ffc6976b5fc3fb4acb325f5413e)
---
*Duration*: 18 minutes and 1 second
*Result*: All 227 tests passed, but please **fix 3 warnings**.
*Test Coverage*: 63%

**Comment by sojanpr** [comment]
👍

**Pull request merged**